src/tools/s3select.js

```diff
@@ -0,0 +1,50 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const nb_native = require('../util/nb_native');
+const { Transform } = require('readable-stream');
+
+class S3SelectStream extends Transform {
+
+    constructor(opts = {}) {
+        super(opts);
+        const context = {
+            query: process.argv[2],
+            input_format: 'CSV',
```

Comment:
Use `argv = require('minimist')` (search for code examples in the repo - there are many) and then you can send argv.query, argv.input_format, argv.field_delimiter etc... as flags (--query --input_format --field_delimiter) to test different options.